torchbenchmark/score/compute_score.py

```diff
@@ -7,32 +7,137 @@
+                self.weights[task] = domain_weight * task_weight * benchmark_weight
+
+    def _setup_benchmark_norms(self):
+        """
+        Helper function which gets the normalization values per benchmark
+        by going through the reference data file.
+        """
+        if not self.ref_data: return
+        elif self.ref_data in [TORCHBENCH_V0_REF_DATA, TORCHBENCH_V1_REF_DATA]:
+            if Path(self.ref_data).exists():
+                with open(self.ref_data) as ref_file:
+                    ref = yaml.full_load(ref_file)
+                self.norm = {b: ref['benchmarks'][b]['norm'] for b in ref['benchmarks']}
+            else:
+                raise ValueError ("File {file} doesn't exist ".format(file=self.ref_data))
+        else:
+            self.norm = {b['name']: b['stats']['mean'] for b in self.ref_data['benchmarks']}
+
+    def get_score_per_config(self, data):
+        """
+        This function iterates over found benchmark dictionary
+        and calculates the weight_sum and benchmark_score.
+        A score_db is then constructed to calculate the cumulative
+        score per config. Here config refers to device, mode and test
+        configurations the benchmark was run on.
+
+        For eg., if the benchmark was run in eval mode on a GPU in Torchscript JIT,
+                    config = (train, cuda, jit)
+
+        This helper returns the score_db .
+
+        """
+        found_benchmarks = defaultdict(lambda: defaultdict(list))
+        score_db = defaultdict(float)
+
+        # Construct a benchmark database by going over through the data file
+        # for the run and update the dictionary by task and model_name
+        for b in data['benchmarks']:
+            name, mean = b['name'], b['stats']['mean']
+            # Extract model_name, test, device and mode from the benchmark name
+            test, model_name, device, mode = re.match(r"test_(.*)\[(.*)\-(.*)\-(.*)\]", name).groups()
+            config = (test, device, mode)
+            task = _get_model_task(model_name)
+            # Append the tuple(mean, config, model_name) for all the configs the
+            # benchmark was run with.
+            found_benchmarks[task][model_name].append((mean, config, name))
+
+        for task, models in found_benchmarks.items():
+            for name, all_configs in models.items():
+                weight = self.weights[task] * (1.0/len(all_configs))
+                for mean, config, benchmark in all_configs:
+                    benchmark_score = weight * math.log(self.norm[benchmark] / mean)
+                    score_db[config] += benchmark_score
+
+        return score_db
+
+    def compute_score(self, data):
+        """
+        This API calculates the total score for all the benchmarks
+        that was run  by reading the data (.json) file.
+        The weights are then calibrated to the target score.
+        """
+        if not self.ref_data:
```

Comment:
pretty unclear why you would want to do this: wouldn't you always get a score of TARGET_SCORE if you use data as ref_data? I would argue that if someone ends up on this codepath, they were using it wrong.  lmk if you had a particular use case in mind.

Reply:
So you would only initialize the ref_data once. So the first file in the data dir will be used as the ref_data and the score for that data will be `TARGET_SCORE` and rest of the scores will be compared against that.  The idea was to initialize the ref_data to first file in the data directory incase no ref_data was provided.